pkg/ddl/reorg.go

```diff
@@ -319,13 +319,15 @@ func extractElemIDs(r *reorgInfo) []int64 {
 
 func (w *worker) mergeWarningsIntoJob(job *model.Job) {
 	rc := w.getReorgCtx(job.ID)
-	rc.mu.Lock()
-	partWarnings := rc.mu.warnings
-	partWarningsCount := rc.mu.warningsCount
-	rc.mu.Unlock()
-	warnings, warningsCount := job.GetWarnings()
-	warnings, warningsCount = mergeWarningsAndWarningsCount(partWarnings, warnings, partWarningsCount, warningsCount)
-	job.SetWarnings(warnings, warningsCount)
+	if rc != nil {
+		rc.mu.Lock()
+		partWarnings := rc.mu.warnings
+		partWarningsCount := rc.mu.warningsCount
+		rc.mu.Unlock()
+		warnings, warningsCount := job.GetWarnings()
+		warnings, warningsCount = mergeWarningsAndWarningsCount(partWarnings, warnings, partWarningsCount, warningsCount)
+		job.SetWarnings(warnings, warningsCount)
+	}
```

Comment:
Do we add logs about `rc == nil` to help with post-processing to avoid 2 reorg workers running concurrently

Reply:
Besides, after https://github.com/pingcap/tidb/pull/52315, could we remove this check?

Reply:
> Besides, after #52315, could we remove this check?

+1, there are many usages assume `rc` is only accessed by single thread.